model/nh_utils.F90

```diff
@@ -2313,20 +2327,27 @@ subroutine nh_BC_k(ptop, grav, kappa, cp, delp, delzBC_t0, delzBC_t1, pt, phis,
    real, dimension(istart:iend, npz-1) :: g_rat
    real, dimension(istart:iend) :: bet
    real :: pm, delz_int
-
+   real, dimension(isd:ied,jsd:jed,npz) :: rgrav, rdg
 
    real :: pealn, pebln, rpkz
 #ifdef MULTI_GASES
       real gamax
 #endif
-   rgrav = 1./grav
    gama = 1./(1.-kappa)
    ptk = ptop ** kappa
    rkap = 1./kappa
    peln1 = log(ptop)
-   rdg = - rdgas * rgrav
    denom = 1./BC_split
 
+   do k=1,npz
+     do j=jsd,jed
+       do i=isd,ied
+         rgrav(i,j,k) = 1./grav_var(i,j,k)
+         rdg(i,j,k) = -rdgas/grav_var(i,j,k)
```

Comment:
This line can be replaced with `rdg(i,j,k) = -rdgas*rgrav(i,j,k)` to avoid divisions.